### ISSUE TITLE
Add "Achievements"

### ISSUE THREAD

**Issue by Volte6** [Volte6/GoMud#32]
# Description

Add passive "Achievements" tracking/reporting.

Achievements shouldn't just be repetition, although some is okay (kill 1000 mobs etc)

# Considerations

* Do <x> without getting hit
* Pick a lock without making a mistake

**Comment by Volte6** [comment]
More thoughts -

Should achievements be known before they are achieved? It's definitely hard to do some without k owing a out them (complete x quest without getting into combat), and once found these sorts of secrets get cataloged somewhere anyways... so why create extra steps for players to search for guides?

On the other hand, certain special achievements being a surprise badge is cool too.

**Comment by windu-ant** [comment]
In this day and age of the internet I think it's bad to keep them a total secret. Like you mentioned people will just look it up. 

I think having a way to hide them though would be useful. Even if just for spoiler reasons. Imagine a player is doing a "main quest" (if you have an overarching one) of your MUD and they see a spoiler for part of it. Perhaps if a player wants to be spoiled they could provide an extra argument: 'achieve list spoil 5' or whichever. It also probably wouldn't be too hard on the coders end to completely block these if they choose.

However, if you also have one that is "kill 1000 goblins" I feel like there's a good reason to provide that task to a player. 

Back in the day a lot of muds were anti quest sharing and punished people for doing it. Most I know of today take a no spoiling stance to it. If someone asks for help you can provide it privately but the idea is you don't ruin it publicly. Even back in the day it was kind of an archaic rule as soon as AIM and MSN messenger existed.

**Comment by Volte6** [comment]
Yes, I think I agree. Having mechanisms built in to avoid players having to search for guides or feel like they are cheating is a good thing.

Quests have an optional "hint" that can be built into them... although I haven't decided how to reveal those hints. I may create a command for it such as:
`quests hint king's shadow` - and the hint for the particular step of the quest they are on will be provided.
I should probably track how many hints were provided for each quest as well, that will provide a little bit of pressure to avoid hints.

For "Achievements" I guess we can just list them all, and let players tackle them as they please.
